coopiteasy_custom/models/sale_timesheet.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models, _
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    @api.model
+    def reset_so_line(self):
+        lines = self.search([])
```

Comment:
maybe just take into account line from open SO or uninvoiced lines would be better than recomputing analytic lines for invoiced lines ?

Reply:
done, check this out.